### PR TITLE
Add unit test for services

### PR DESCRIPTION
**Pull request by GhiocelAndrei**
Solves issue #22.

### CODE REVIEW

File: IssueTracker.Testing/ServicesTest/IssueServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            // Act
+            var result = await _issueService.GetAll();
+
+            // Assert
+            _mockRepository.Verify(x => x.GetAllAsync(), Times.Once);
+            Assert.Equal(2, result.Count());
+        }
+
+        [Fact]
+        public async Task Get_ById_ShouldReturnIssue()
```
Comment (cristi-paraschiv) [inline IssueTracker.Testing/ServicesTest/IssueServiceTests.cs:40]: you should also add a test that checks service returns null for non-existing issue

File: IssueTracker.Testing/IssueTracker.Testing.csproj
Comment (cristi-paraschiv) [inline IssueTracker.Testing/IssueTracker.Testing.csproj:1]: we should have one test project per application project

File: IssueTracker.Testing/ServicesTest/IssueServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+        [Fact]
+        public async Task GetAll_ShouldReturnAllIssues()
+        {
+            // Arrange
+            var issues = new List<Issue> { new Issue(), new Issue() };
+            _mockRepository.Setup(x => x.GetAllAsync()).ReturnsAsync(issues);
+
+            // Act
+            var result = await _issueService.GetAll();
```
Comment (florincatrina) [inline IssueTracker.Testing/ServicesTest/IssueServiceTests.cs:32]: Async

File: IssueTracker.Testing/ServicesTest/IssueServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+            // Assert
+            _mockRepository.Verify(x => x.GetAllAsync(), Times.Once);
+            Assert.Equal(2, result.Count());
```
Comment (florincatrina) [inline IssueTracker.Testing/ServicesTest/IssueServiceTests.cs:36]: Could use Fluent Assertions

File: IssueTracker.Testing/ServicesTest/IssueServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+            });
+
+            _mockRepository.Setup(r => r.UpdateAsync(It.IsAny<Issue>())).ReturnsAsync((Issue i) => i);
+
+            // Act
+            var result = await _issueService.Update(1, updateIssueCommand);
+
+            // Assert
+            _mockRepository.Verify(r => r.GetAsync(1), Times.Once);
+            _mockMapper.Verify(m => m.Map(updateIssueCommand, issue), Times.Once);
```
Comment (florincatrina) [inline IssueTracker.Testing/ServicesTest/IssueServiceTests.cs:100]: I wouldn't verify mapping, checking the actual property values is enough. Checking repository makes sense as this is an external dependency for your sut.

File: IssueTracker.Testing/ServicesTest/IssueServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+                Title = "New title",
+                Description = "New description",
+                Priority = Priority.Low,
+                ReporterId = 2,
+                AssigneeId = 2,
+                ProjectId = 2
+            };
+
+            _mockRepository.Setup(r => r.GetAsync(1)).ReturnsAsync(issue);
+            _mockMapper.Setup(m => m.Map(updateIssueCommand, issue)).Callback((UpdateIssueCommand source, Issue target) => {
```
[the remark this answers is not in the document]
Comment (GhiocelAndrei) [reply IssueTracker.Testing/ServicesTest/IssueServiceTests.cs]: Changed mocked mapper to real mapper used by application

File: IssueTracker.Testing/ServicesTest/IssueServiceTests.cs (hunk 2 above)
Comment (GhiocelAndrei) [reply IssueTracker.Testing/ServicesTest/IssueServiceTests.cs]: I created Issue #25 and I will add 'Async' suffix to all async methods in a later PR.

File: IssueTracker.Testing/ServicesTest/IssueServiceTests.cs (hunk 3 above)
Comment (GhiocelAndrei) [reply IssueTracker.Testing/ServicesTest/IssueServiceTests.cs]: I will look into them.

**Review by cristi-paraschiv — APPROVED** [review]
